### ISSUE TITLE
Use better on-screen keyboard

### ISSUE THREAD

**Issue by SomeoneToIgnore** [m-weigand/pinenote-debian-recipes#25]
Landscape mode, current rootfs keyboard:
![linux_keyboard](https://user-images.githubusercontent.com/2690773/207323553-9f9e6761-e6aa-47da-bcef-3d742e0d8632.png)

Landscape mode, Android keyboard (top overlay with ESC, arrows and other symbols comes from termux, not the keyboard per se):
![android_keyboard](https://user-images.githubusercontent.com/2690773/207323564-6c213520-52c4-4e10-a96e-3eb38aa1d100.png)

Current Gnome on-screen keyboard is very hard to use:
* it's relatively slow to draw on e-ink (esp. the pressing animations) due to being very dark 
* there's plenty of ghosting on the keys
* the distance between keys is very small, same as the keys themselves
* the digits/special characters are relatively inconvenient to turn off: `?..` button is not on the edge of the keyboard as on almost every modern mobile keyboard
* the digits/special characters are *very* inconvenient to input: the keyboard toggles back to letters after every digit/symbol input, so the default `1234` user password has to use 8 symbols that are relatively slow due to redrawing dark animations being slow

**Comment by m-weigand** [comment]
I agree, the keyboard is sometimes really unusable. However, I'm still not really sure what to do about it. I also think there are at least two onscreen-keyboard variants in gnome, with the new one being caribou, https://gitlab.gnome.org/GNOME/caribou. It says there "configurable", but I do not understand how to actually change anything at runtime...

**Comment by m-weigand** [comment]
The numbers-do-not-latch issue is tracked here: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5763

**Comment by m-weigand** [comment]
There exists an extension to improve the osk, but a rewrite for gnome 43 is still in progress, see: https://github.com/nick-shmyrev/improved-osk-gnome-ext/issues/30

Still, there apparently is a wip branch working on gnome 43

**Comment by m-weigand** [comment]
Not fully related, but for later reference: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/6238

**Comment by m-weigand** [comment]
This was mentioned in the chat. We should revisit the options:

* https://github.com/Vishram1123/gjs-osk (was recommended in a reddit post and seems to work in gnome 47)
* https://github.com/maliit/keyboard (not sure how to activate this in Gnome)
* https://github.com/nick-shmyrev/improved-osk-gnome-ext (as far as I can see, Gnome 47 is not supported)

**Comment by WilsontheWolf** [comment]
Figured I would play with some of these. I was able to install gjs-osk.You need to install the `gnome-browser-connector` package to install extensions from the site, and run `gsettings set org.gnome.shell disable-extension-version-validation "true"` to disable the version check. I found it was a bit small by default but I switched the layout to Tenkeyless Internation (Compact Internation also should work), and you can adjust the size in the settings to how you would like). It doesn't seem to push up apps like the gnome one does, but can be dragged around to work around that. I'll use it for a bit and see how it works.
